### ISSUE TITLE
radio.receive(str) blocks forever with Arduino BLE Sense and CC1101

### ISSUE THREAD

**Issue by lasryaric** [jgromes/RadioLib#277]
**Describe the bug**
radio.receive(str) blocks for ever...
```
  int state = radio.receive(str);
```

**To Reproduce**
CC1101 receiver sample code on Arduino BLE sense 33

```
/*
   RadioLib CC1101 Receive Example

   This example receives packets using CC1101 FSK radio module.
   To successfully receive data, the following settings have to be the same
   on both transmitter and receiver:
    - carrier frequency
    - bit rate
    - frequency deviation
    - sync word

   For default module settings, see the wiki page
   https://github.com/jgromes/RadioLib/wiki/Default-configuration#cc1101

   For full API reference, see the GitHub Pages
   https://jgromes.github.io/RadioLib/
*/

//// include the library
#include <RadioLib.h>
//
//// CC1101 has the following connections:
//// CS pin:    10
//// GDO0 pin:  2
//// RST pin:   unused
//// GDO2 pin:  3 (optional)
CC1101 radio = new Module(10, 2, RADIOLIB_NC, 3);
//
//// or using RadioShield
//// https://github.com/jgromes/RadioShield
////CC1101 radio = RadioShield.ModuleA;

void setup() {
  Serial.begin(9600);
  while (!Serial) ;
  Serial.println("serial opened!!!");
  // initialize CC1101 with default settings
//  
  int state = radio.begin();
  if (state == ERR_NONE) {
    Serial.println("success! 2");
  } 
  else {
    Serial.print(F("failed, code "));
    Serial.println(state);
    while (true);
  }
}

void loop() {
  
  Serial.print(F("[CC1101] Waiting for incoming transmission ... \n"));
  delay(1000);

  // you can receive data as an Arduino String
  String str;
  int state = radio.receive(str);
  
  Serial.print(F("[CC1101] Received!\n"));

  if (state == ERR_NONE) {
    // packet was successfully received
    Serial.println(F("success!"));

    // print the data of the packet
    Serial.print(F("[CC1101] Data:\t\t"));
    Serial.println(str);

    // print RSSI (Received Signal Strength Indicator)
    // of the last received packet
    Serial.print(F("[CC1101] RSSI:\t\t"));
    Serial.print(radio.getRSSI());
    Serial.println(F(" dBm"));

    // print LQI (Link Quality Indicator)
    // of the last received packet, lower is better
    Serial.print(F("[CC1101] LQI:\t\t"));
    Serial.println(radio.getLQI());

  } else if (state == ERR_CRC_MISMATCH) {
    // packet was received, but is malformed
    Serial.println(F("CRC error!"));

  } else {
    // some other error occurred
    Serial.print(F("failed, code "));
    Serial.println(state);

  }

}
```

**Expected behavior**
`loop()` function should keep iterating and print the received data.

**Screenshots**
N/A

**Additional info (please complete):**
 - MCU: Arduino BLE sense 33 nano

 - Wireless module type CC1101 ( https://www.amazon.com/SMAKN%C2%AE-Wireless-Technical-External-Antenna/dp/B00U5TO37W )
 - Arduino IDE version  [e.g. 1.8.13]
 - Library version: radiolib 4.3.0

Any help in understanding why the function blocks forever would be really appreciated. I am an experienced software engineer but I have 0 knowledge in Arduino.

Thanks.

**Comment by lasryaric** [comment]
I should add that it does fork for a few iteration, the `radio.receive(str)` function returns some data, but after a few iteration (~5) it starts blocking forever.

**Comment by jgromes** [comment]
As far as I can tell, nothing in `CC1101::receive()` has the potential to be an infinite loop, all while cycles that wait for pin changes are guarded with timeouts. Unforunately I don't have a BLE Sense 33 Nano on hand to try it out. You could try running in [debug verbose](https://github.com/jgromes/RadioLib/wiki/Debug-mode), to see what's the last SPI exchange that is performed.

**Comment by lasryaric** [comment]
Here is the output I get with the debug and verbose mode enabled.
The last line showing up in the logs is `14:54:33.680 -> [CC1101] Waiting for incoming transmission ... `, which is right before `radio.receive(str)`.

Any idea how about how to make progress from here?

Thanks


```
14:39:00.658 -> serial opened!!!
14:39:00.725 -> success! 2
14:39:00.725 -> [CC1101] Waiting for incoming transmission ... 
14:39:02.233 -> [CC1101] Received!
14:39:02.233 -> failed, code -6
14:39:02.233 -> [CC1101] Waiting for incoming transmission ... 
14:39:03.738 -> [CC1101] Received!
14:39:03.738 -> failed, code -6
14:39:03.738 -> [CC1101] Waiting for incoming transmission ... 
14:39:05.243 -> [CC1101] Received!
14:39:05.243 -> failed, code -6
14:39:05.243 -> [CC1101] Waiting for incoming transmission ... 
14:39:06.399 -> [CC1101] Received!
14:39:06.399 -> CRC error!
14:39:06.399 -> [CC1101] Waiting for incoming transmission ... 
14:39:07.923 -> [CC1101] Received!
14:39:07.923 -> failed, code -6
14:39:07.923 -> [CC1101] Waiting for incoming transmission ... 
14:39:09.348 -> [CC1101] Received!
14:39:09.348 -> CRC error!
14:39:09.348 -> [CC1101] Waiting for incoming transmission ... 
14:40:37.742 -> serial opened!!!
14:40:37.885 -> success! 2
14:40:37.885 -> [CC1101] Waiting for incoming transmission ... 
14:40:39.371 -> [CC1101] Received!
14:40:39.371 -> failed, code -6
14:40:39.371 -> [CC1101] Waiting for incoming transmission ... 
14:40:40.891 -> [CC1101] Received!
14:40:40.891 -> failed, code -6
14:40:40.891 -> [CC1101] Waiting for incoming transmission ... 
14:40:42.398 -> [CC1101] Received!
14:40:42.398 -> failed, code -6
14:40:42.398 -> [CC1101] Waiting for incoming transmission ... 
14:40:43.850 -> [CC1101] Received!
14:40:43.850 -> CRC error!
14:40:43.850 -> [CC1101] Waiting for incoming transmission ... 
14:40:45.354 -> [CC1101] Received!
14:40:45.354 -> failed, code -6
14:40:45.354 -> [CC1101] Waiting for incoming transmission ... 
14:40:46.853 -> [CC1101] Received!
14:40:46.853 -> failed, code -6
14:40:46.853 -> [CC1101] Waiting for incoming transmission ... 
14:48:00.955 -> serial opened!!!
14:48:01.093 -> success! 2
14:48:01.093 -> [CC1101] Waiting for incoming transmission ... 
14:48:02.609 -> [CC1101] Received!
14:48:02.609 -> failed, code -6
14:48:02.609 -> [CC1101] Waiting for incoming transmission ... 
14:48:04.093 -> [CC1101] Received!
14:48:04.093 -> failed, code -6
14:48:04.093 -> [CC1101] Waiting for incoming transmission ... 
14:48:05.593 -> [CC1101] Received!
14:48:05.593 -> failed, code -6
14:48:05.593 -> [CC1101] Waiting for incoming transmission ... 
14:48:07.106 -> [CC1101] Received!
14:48:07.106 -> failed, code -6
14:48:07.106 -> [CC1101] Waiting for incoming transmission ... 
14:48:08.611 -> [CC1101] Received!
14:48:08.611 -> failed, code -6
14:48:08.611 -> [CC1101] Waiting for incoming transmission ... 
14:48:09.816 -> [CC1101] Received!
14:48:09.816 -> CRC error!
14:48:09.816 -> [CC1101] Waiting for incoming transmission ... 
14:53:52.455 -> serial opened!!!
14:53:52.455 -> R	71	14	
14:53:52.455 -> M	CC1101
14:53:52.601 -> R	18	4	
14:53:52.601 -> W	18	14	
14:53:52.601 -> R	18	14	
14:53:52.601 -> R	7	4	
14:53:52.601 -> W	7	4	
14:53:52.601 -> R	7	4	
14:53:52.601 -> R	8	45	
14:53:52.601 -> W	8	5	
14:53:52.601 -> R	8	5	
14:53:52.601 -> R	8	5	
14:53:52.601 -> W	8	5	
14:53:52.601 -> R	8	5	
14:53:52.601 -> R	D	1E	
14:53:52.601 -> W	D	10	
14:53:52.601 -> R	D	10	
14:53:52.601 -> R	E	C4	
14:53:52.601 -> W	E	B1	
14:53:52.601 -> R	E	B1	
14:53:52.601 -> R	F	EC	
14:53:52.601 -> W	F	3B	
14:53:52.601 -> R	F	3B	
14:53:52.601 -> R	7E	C6	
14:53:52.634 -> W	7E	60	
14:53:52.634 -> R	7E	60	
14:53:52.634 -> R	10	8C	
14:53:52.634 -> W	10	8A	
14:53:52.634 -> R	10	8A	
14:53:52.634 -> R	11	22	
14:53:52.634 -> W	11	E3	
14:53:52.634 -> R	11	E3	
14:53:52.634 -> R	10	8A	
14:53:52.634 -> W	10	AA	
14:53:52.634 -> R	10	AA	
14:53:52.634 -> R	15	47	
14:53:52.634 -> W	15	47	
14:53:52.634 -> R	15	47	
14:53:52.634 -> R	15	47	
14:53:52.634 -> W	15	47	
14:53:52.634 -> R	15	47	
14:53:52.634 -> R	7E	60	
14:53:52.634 -> W	7E	C0	
14:53:52.634 -> R	7E	C0	
14:53:52.634 -> R	8	5	
14:53:52.634 -> W	8	5	
14:53:52.634 -> R	8	5	
14:53:52.634 -> R	6	FF	
14:53:52.634 -> W	6	3F	
14:53:52.634 -> R	6	3F	
14:53:52.634 -> R	13	22	
14:53:52.634 -> W	13	2	
14:53:52.634 -> R	13	2	
14:53:52.634 -> R	12	2	
14:53:52.634 -> W	12	2	
14:53:52.634 -> R	12	2	
14:53:52.634 -> R	12	2	
14:53:52.634 -> W	12	2	
14:53:52.634 -> R	12	2	
14:53:52.634 -> R	8	5	
14:53:52.634 -> W	8	5	
14:53:52.634 -> R	8	5	
14:53:52.634 -> R	12	2	
14:53:52.634 -> W	12	2	
14:53:52.634 -> R	12	2	
14:53:52.634 -> R	4	D3	
14:53:52.634 -> W	4	12	
14:53:52.634 -> R	4	12	
14:53:52.634 -> R	5	91	
14:53:52.669 -> W	5	AD	
14:53:52.669 -> R	5	AD	
14:53:52.669 -> success! 2
14:53:52.669 -> [CC1101] Waiting for incoming transmission ... 
14:53:53.662 -> R	2	3F	
14:53:53.662 -> W	2	6	
14:53:53.662 -> R	2	6	
14:53:53.696 -> R	7F	0	
14:53:53.696 -> R	7	4	
14:53:53.696 -> R	7F	
14:53:53.696 -> R	7F	97	
14:53:53.696 -> R	7F	7D	
14:53:53.696 -> [CC1101] Received!
14:53:53.696 -> CRC error!
14:53:53.696 -> [CC1101] Waiting for incoming transmission ... 
14:53:54.683 -> R	2	6	
14:53:54.683 -> W	2	6	
14:53:54.683 -> R	2	6	
14:53:55.207 -> [CC1101] Received!
14:53:55.207 -> failed, code -6
14:53:55.207 -> [CC1101] Waiting for incoming transmission ... 
14:53:56.207 -> R	2	6	
14:53:56.207 -> W	2	6	
14:53:56.207 -> R	2	6	
14:53:56.716 -> [CC1101] Received!
14:53:56.716 -> failed, code -6
14:53:56.716 -> [CC1101] Waiting for incoming transmission ... 
14:53:57.707 -> R	2	6	
14:53:57.707 -> W	2	6	
14:53:57.707 -> R	2	6	
14:53:58.021 -> R	7F	13	
14:53:58.021 -> R	7	4	
14:53:58.021 -> R	7F	7C	A	FE	9B	55	23	14	2F	F0	CB	8C	AD	66	80	FA	97	15	63	70	
14:53:58.021 -> R	7F	F1	
14:53:58.021 -> R	7F	7F	
14:53:58.021 -> [CC1101] Received!
14:53:58.021 -> CRC error!
14:53:58.021 -> [CC1101] Waiting for incoming transmission ... 
14:53:59.020 -> R	2	6	
14:53:59.020 -> W	2	6	
14:53:59.020 -> R	2	6	
14:53:59.447 -> R	7F	2E	
14:53:59.447 -> R	7	4	
14:53:59.447 -> R	7F	49	EF	10	52	D8	15	8B	36	1	FC	A0	2A	58	E0	D4	EE	D2	FC	84	9A	55	5F	C	C6	55	32	14	50	7	59	63	11	5	4	FC	DF	F4	CF	F3	95	1	3	4	B7	9F	90	
14:53:59.447 -> R	7F	FC	
14:53:59.447 -> R	7F	7F	
14:53:59.447 -> [CC1101] Received!
14:53:59.447 -> CRC error!
14:53:59.447 -> [CC1101] Waiting for incoming transmission ... 
14:54:00.466 -> R	2	6	
14:54:00.466 -> W	2	6	
14:54:00.466 -> R	2	6	
14:54:00.963 -> [CC1101] Received!
14:54:00.963 -> failed, code -6
14:54:00.963 -> [CC1101] Waiting for incoming transmission ... 
14:54:01.944 -> R	2	6	
14:54:01.944 -> W	2	6	
14:54:01.944 -> R	2	6	
14:54:02.451 -> [CC1101] Received!
14:54:02.451 -> failed, code -6
14:54:02.451 -> [CC1101] Waiting for incoming transmission ... 
14:54:03.448 -> R	2	6	
14:54:03.448 -> W	2	6	
14:54:03.448 -> R	2	6	
14:54:03.955 -> [CC1101] Received!
14:54:03.955 -> failed, code -6
14:54:03.955 -> [CC1101] Waiting for incoming transmission ... 
14:54:04.956 -> R	2	6	
14:54:04.956 -> W	2	6	
14:54:04.956 -> R	2	6	
14:54:05.463 -> [CC1101] Received!
14:54:05.463 -> failed, code -6
14:54:05.463 -> [CC1101] Waiting for incoming transmission ... 
14:54:06.478 -> R	2	6	
14:54:06.478 -> W	2	6	
14:54:06.478 -> R	2	6	
14:54:06.962 -> [CC1101] Received!
14:54:06.962 -> failed, code -6
14:54:06.962 -> [CC1101] Waiting for incoming transmission ... 
14:54:07.955 -> R	2	6	
14:54:07.955 -> W	2	6	
14:54:07.955 -> R	2	6	
14:54:08.457 -> [CC1101] Received!
14:54:08.457 -> failed, code -6
14:54:08.457 -> [CC1101] Waiting for incoming transmission ... 
14:54:09.475 -> R	2	6	
14:54:09.475 -> W	2	6	
14:54:09.475 -> R	2	6	
14:54:09.973 -> [CC1101] Received!
14:54:09.973 -> failed, code -6
14:54:09.973 -> [CC1101] Waiting for incoming transmission ... 
14:54:10.978 -> R	2	6	
14:54:10.978 -> W	2	6	
14:54:10.978 -> R	2	6	
14:54:11.470 -> [CC1101] Received!
14:54:11.470 -> failed, code -6
14:54:11.470 -> [CC1101] Waiting for incoming transmission ... 
14:54:12.493 -> R	2	6	
14:54:12.493 -> W	2	6	
14:54:12.493 -> R	2	6	
14:54:12.672 -> R	7F	7F	
14:54:12.672 -> R	7	4	
14:54:12.672 -> R	7F	49	EF	10	52	D8	15	8B	36	1	FC	A0	2A	58	E0	D4	EE	D2	FC	84	9A	55	5F	C	C6	55	32	14	50	7	59	63	11	5	4	FC	DF	F4	CF	F3	95	1	3	4	B7	9F	90	FC	7F	80	CE	86	7B	4F	19	E3	F9	35	87	28	A2	93	DC	39	D8	49	EF	10	52	D8	15	8B	36	1	FC	A0	2A	58	E0	D4	EE	D2	FC	84	9A	55	5F	C	C6	55	32	14	50	7	59	63	11	5	4	FC	DF	F4	CF	F3	95	1	3	4	B7	9F	90	FC	7F	80	CE	86	7B	4F	19	E3	F9	35	87	28	A2	93	DC	39	
14:54:12.706 -> R	7F	39	
14:54:12.706 -> R	7F	D8	
14:54:12.706 -> [CC1101] Received!
14:54:12.706 -> success!
14:54:12.706 -> [CC1101] Data:		I⸮R⸮⸮6⸮⸮*X⸮⸮⸮⸮⸮⸮⸮U_⸮U2PYc⸮⸮⸮⸮⸮⸮⸮⸮⸮⸮Ά{O⸮⸮5⸮(⸮⸮⸮9⸮I⸮R⸮⸮]O⸮⸮7r~⸮⸮qt⸮⸮@⸮t+⸮⸮⸮o⸮⸮Ά{O⸮⸮5⸮(⸮⸮⸮9ؗ}⸮U8ˬ;⸮⸮v
14:54:12.706 -> [CC1101] RSSI:		-45.50 dBm
14:54:12.706 -> [CC1101] LQI:		88
14:54:12.706 -> [CC1101] Waiting for incoming transmission ... 
14:54:13.719 -> R	2	6	
14:54:13.719 -> W	2	6	
14:54:13.719 -> R	2	6	
14:54:14.211 -> [CC1101] Received!
14:54:14.211 -> failed, code -6
14:54:14.211 -> [CC1101] Waiting for incoming transmission ... 
14:54:15.203 -> R	2	6	
14:54:15.203 -> W	2	6	
14:54:15.203 -> R	2	6	
14:54:15.687 -> R	7F	49	
14:54:15.687 -> R	7	4	
14:54:15.687 -> R	7F	49	EF	10	52	D8	15	8B	36	1	FC	A0	2A	58	E0	D4	EE	D2	FC	84	9A	55	5F	C	C6	55	32	14	50	7	59	63	11	5	4	FC	DF	F4	CF	F3	95	1	3	4	B7	9F	90	FC	7F	80	CE	86	7B	4F	19	E3	F9	35	87	28	A2	93	DC	39	D8	49	EF	10	52	D8	15	8B	36	1	
14:54:15.724 -> R	7F	FC	
14:54:15.724 -> R	7F	A0	
14:54:15.724 -> [CC1101] Received!
14:54:15.724 -> success!
14:54:15.724 -> [CC1101] Data:		I⸮R⸮⸮6⸮⸮*X⸮⸮⸮⸮⸮⸮⸮U_⸮U2PYc⸮⸮⸮⸮⸮⸮⸮⸮⸮⸮Ά{O⸮⸮5⸮(⸮⸮⸮9⸮I⸮RY⸮]O
14:54:15.724 -> [CC1101] RSSI:		-76.00 dBm
14:54:15.724 -> [CC1101] LQI:		32
14:54:15.724 -> [CC1101] Waiting for incoming transmission ... 
14:54:16.714 -> R	2	6	
14:54:16.714 -> W	2	6	
14:54:16.714 -> R	2	6	
14:54:17.200 -> [CC1101] Received!
14:54:17.200 -> failed, code -6
14:54:17.200 -> [CC1101] Waiting for incoming transmission ... 
14:54:18.215 -> R	2	6	
14:54:18.215 -> W	2	6	
14:54:18.215 -> R	2	6	
14:54:18.711 -> [CC1101] Received!
14:54:18.711 -> failed, code -6
14:54:18.711 -> [CC1101] Waiting for incoming transmission ... 
14:54:19.723 -> R	2	6	
14:54:19.723 -> W	2	6	
14:54:19.723 -> R	2	6	
14:54:20.037 -> R	7F	2A	
14:54:20.037 -> R	7	4	
14:54:20.037 -> R	7F	49	EF	10	52	D8	15	8B	36	1	FC	A0	2A	58	E0	D4	EE	D2	FC	84	9A	55	5F	C	C6	55	32	14	50	7	59	63	11	5	4	FC	DF	F4	CF	F3	95	1	3	
14:54:20.074 -> R	7F	4	
14:54:20.074 -> R	7F	B7	
14:54:20.074 -> [CC1101] Received!
14:54:20.074 -> success!
14:54:20.074 -> [CC1101] Data:		I⸮R⸮⸮6⸮⸮*X⸮⸮⸮⸮⸮⸮⸮U_⸮U2PYc⸮⸮⸮⸮⸮
14:54:20.074 -> [CC1101] RSSI:		-72.00 dBm
14:54:20.074 -> [CC1101] LQI:		55
14:54:20.074 -> [CC1101] Waiting for incoming transmission ... 
14:54:21.082 -> R	2	6	
14:54:21.082 -> W	2	6	
14:54:21.082 -> R	2	6	
14:54:21.119 -> R	7F	9F	
14:54:21.119 -> R	7	4	
14:54:21.119 -> R	7F	49	EF	10	52	D8	15	8B	36	1	FC	A0	2A	58	E0	D4	EE	D2	FC	84	9A	55	5F	C	C6	55	32	14	50	7	59	63	11	5	4	FC	DF	F4	CF	F3	95	1	3	4	B7	9F	90	FC	7F	80	CE	86	7B	4F	19	E3	F9	35	87	28	A2	93	DC	39	D8	49	EF	10	52	D8	15	8B	36	1	FC	A0	2A	58	E0	D4	EE	D2	FC	84	9A	55	5F	C	C6	55	32	14	50	7	59	63	11	5	4	FC	DF	F4	CF	F3	95	1	3	4	B7	9F	90	FC	7F	80	CE	86	7B	4F	19	E3	F9	35	87	28	A2	93	DC	39	39	D8	49	EF	10	52	D8	15	8B	36	1	FC	A0	2A	58	E0	D4	EE	D2	FC	84	9A	55	5F	C	C6	55	32	14	50	7	59	
14:54:21.153 -> R	7F	63	
14:54:21.153 -> R	7F	11	
14:54:21.153 -> [CC1101] Received!
14:54:21.153 -> CRC error!
14:54:21.153 -> [CC1101] Waiting for incoming transmission ... 
14:54:22.176 -> R	2	6	
14:54:22.176 -> W	2	6	
14:54:22.176 -> R	2	6	
14:54:22.668 -> [CC1101] Received!
14:54:22.668 -> failed, code -6
14:54:22.668 -> [CC1101] Waiting for incoming transmission ... 
14:54:23.661 -> R	2	6	
14:54:23.661 -> W	2	6	
14:54:23.661 -> R	2	6	
14:54:24.050 -> R	7F	5	
14:54:24.050 -> R	7	4	
14:54:24.050 -> R	7F	1A	AC	EE	3F	1F	
14:54:24.050 -> R	7F	E1	
14:54:24.050 -> R	7F	EC	
14:54:24.050 -> [CC1101] Received!
14:54:24.050 -> success!
14:54:24.050 -> [CC1101] Data:		⸮⸮?
14:54:24.050 -> [CC1101] RSSI:		-89.50 dBm
14:54:24.050 -> [CC1101] LQI:		108
14:54:24.050 -> [CC1101] Waiting for incoming transmission ... 
14:54:25.063 -> R	2	6	
14:54:25.063 -> W	2	6	
14:54:25.063 -> R	2	6	
14:54:25.552 -> [CC1101] Received!
14:54:25.552 -> failed, code -6
14:54:25.552 -> [CC1101] Waiting for incoming transmission ... 
14:54:26.568 -> R	2	6	
14:54:26.568 -> W	2	6	
14:54:26.568 -> R	2	6	
14:54:27.063 -> [CC1101] Received!
14:54:27.063 -> failed, code -6
14:54:27.063 -> [CC1101] Waiting for incoming transmission ... 
14:54:28.038 -> R	2	6	
14:54:28.038 -> W	2	6	
14:54:28.038 -> R	2	6	
14:54:28.571 -> [CC1101] Received!
14:54:28.571 -> failed, code -6
14:54:28.571 -> [CC1101] Waiting for incoming transmission ... 
14:54:29.554 -> R	2	6	
14:54:29.554 -> W	2	6	
14:54:29.554 -> R	2	6	
14:54:29.696 -> R	7F	54	
14:54:29.696 -> R	7	4	
14:54:29.696 -> R	7F	1A	AC	EE	3F	1F	E1	EC	54	7E	AC	F3	84	60	8D	B3	A2	98	FC	AE	90	7B	E0	60	E2	BF	4D	B1	F7	7F	59	63	11	5	4	FC	DF	F4	CF	F3	95	1	3	4	B7	9F	90	FC	7F	80	CE	86	7B	4F	19	E3	F9	35	87	28	A2	93	DC	39	D8	1A	AC	EE	3F	1F	E1	EC	54	7E	AC	F3	84	60	8D	B3	A2	98	FC	AE	90	
14:54:29.696 -> R	7F	7B	
14:54:29.696 -> R	7F	E0	
14:54:29.696 -> [CC1101] Received!
14:54:29.696 -> success!
14:54:29.696 -> [CC1101] Data:		⸮⸮?⸮⸮T~⸮⸮`⸮⸮⸮⸮⸮⸮⸮{⸮`⸮M⸮⸮Yc⸮⸮⸮⸮⸮⸮⸮⸮⸮⸮Ά{O⸮⸮5⸮(⸮⸮⸮9⸮⸮⸮?i_⸮U2PY6⸮⸮qt
14:54:29.696 -> [CC1101] RSSI:		-12.50 dBm
14:54:29.696 -> [CC1101] LQI:		96
14:54:29.696 -> [CC1101] Waiting for incoming transmission ... 
14:54:30.709 -> R	2	6	
14:54:30.709 -> W	2	6	
14:54:30.709 -> R	2	6	
14:54:30.997 -> R	7F	1A	
14:54:30.997 -> R	7	4	
14:54:30.997 -> R	7F	F6	31	1D	1	6D	51	3F	9C	35	0	EC	2	D1	81	59	F	65	80	49	44	58	7F	E3	C7	FC	A0	
14:54:30.997 -> R	7F	F0	
14:54:30.997 -> R	7F	7F	
14:54:30.997 -> [CC1101] Received!
14:54:30.997 -> CRC error!
14:54:30.997 -> [CC1101] Waiting for incoming transmission ... 
14:54:32.014 -> R	2	6	
14:54:32.014 -> W	2	6	
14:54:32.014 -> R	2	6	
14:54:32.513 -> [CC1101] Received!
14:54:32.513 -> failed, code -6
14:54:32.513 -> [CC1101] Waiting for incoming transmission ... 
14:54:33.512 -> R	2	6	
14:54:33.512 -> W	2	6	
14:54:33.512 -> R	2	6	
14:54:33.646 -> R	7F	7F	
14:54:33.646 -> R	7	4	
14:54:33.646 -> R	7F	F6	31	1D	1	6D	51	3F	9C	35	0	EC	2	D1	81	59	F	65	80	49	44	58	7F	E3	C7	FC	A0	F0	7F	7F	59	63	11	5	4	FC	DF	F4	CF	F3	95	1	3	4	B7	9F	90	FC	7F	80	CE	86	7B	4F	19	E3	F9	35	87	28	A2	93	DC	39	D8	F6	31	1D	1	6D	51	3F	9C	35	0	EC	2	D1	81	59	F	65	80	49	44	58	7F	E3	C7	FC	A0	F0	7F	7F	59	63	11	5	4	FC	DF	F4	CF	F3	95	1	3	4	B7	9F	90	FC	7F	80	CE	86	7B	4F	19	E3	F9	35	87	28	A2	93	DC	39	
14:54:33.680 -> R	7F	39	
14:54:33.680 -> R	7F	D8	
14:54:33.680 -> [CC1101] Received!
14:54:33.680 -> success!
14:54:33.680 -> [CC1101] Data:		
14:54:33.680 -> [CC1101] RSSI:		-45.50 dBm
14:54:33.680 -> [CC1101] LQI:		88
14:54:33.680 -> [CC1101] Waiting for incoming transmission ... 

````

**Comment by jgromes** [comment]
Looks like it gets stuck somewhere around here:

https://github.com/jgromes/RadioLib/blob/86ca714d0017419aebc5bae5b4e8010e71e66874/src/modules/CC1101/CC1101.cpp#L288-L297

Line 296 should print out 
```
14:54:33.512 -> R	2	6	
14:54:33.512 -> W	2	6	
14:54:33.512 -> R	2	6
```

when it gets executed. We don't see that, so taking into account the fact that Serial output has some delay, I'd say it's crashing somewhere around there. There's nothing but SPI transactions there, so the problem isn't obvious.

Digging around on Arduino forums a bit, I found the following: https://forum.arduino.cc/t/nano-33-ble-and-spi-stops-loop-running/683020/2

TL;DR - apparently, Nano 33 BLE will go to sleep mode in `delay()` which is quite a unique behavior and I have no idea what will it do with SPI. Could you try the Receive_Interrupt example to rule this out?

**Comment by jgromes** [comment]
Closing as inactive - AFAIK this only happens on Nano 33 BLE, so I can't reproduce it. Feel free to reopen if there's more information.
